### DISCUSSION TITLE
restart in script fails

### DISCUSSION THREAD

**Discussion by hw-schrauber** [arendst/Tasmota#18929] [Q&A]
Hello forum,

I have compiled a DE version 12.5.0 with scripting to read energy meters. Everything runs so far. Since I have a HW that I want to restart once a day for testing purposes, I added the following at the end of the script:

h=hours
if chg[h]>0
and h==17
then
=>restart 1
endif

This condition is also executed. The only problem is that after executing “restart 1” from the script, scripting is disabled. If I call “restart 1” directly in the console, scripting remains active. It is also HW independent. I have tested it on several HW. Always the same. Is this normal with scripting? Or is this probably a bug?

**Comment by pkkrusty** [comment]
Sounds like you have a boot loop, which disables scripting after a few reboots in a row. The script initializes h=0 in the >D section (maybe?), then assign h=hours, which will still cause the script to reboot as long as the hour is 17.

Maybe better to to just use `upsecs`. Then you can just do 
```
if upsecs>86400
then
=>restart 1
endif
```

**Reply by hw-schrauber** [reply]
Hi pkkrusty,
I have tested your variant with a different time value. 

if upsecs>240
then
=>restart 1
endif

But there is also a phenomenon here. After I have saved the change in the script, the device restarts directly. It takes exactly 2 minutes until the script triggers a restart again. But then every second a restart is triggered.

00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: aus Flash geladen von F4, Anzahl 42
00:00:00.055 QPC: Count 1
00:00:00.074 Projekt tasmota - Energy-SmartMeter Version 12.5.0(tasmota)-2_7_4_9(2023.06.13 12:25:52)
00:00:00.081 Script: nv=4, tv=0, vns=11, vmem=79, smem=2560
00:00:00.082 Script: performs "sensor53 r"
00:00:00.109 SNS: Hardware Serial
00:00:00.110 meters: 1 , decode lines: 3, memory used: 1538 bytes
00:00:00.111 RSL: RESULT = {"Time":"1970-01-01T00:00:00","SML":{"CMD":"restart"}}
00:00:00.117 SNS: Hardware Serial
00:00:00.118 meters: 1 , decode lines: 3, memory used: 1538 bytes
00:00:00.558 WIF: verbinden mit AP2 AP-RS Channel 5 BSSId DC:15:C8:DB:FA:82 in Modus 11n als tasmota-7EB07A-4218...
00:00:01.758 WIF: verbunden
00:00:02.011 HTP: Web-Server aktiv bei tasmota-7EB07A-4218 mit IP-Adresse 192.168.60.119
10:26:32.025 RSL: INFO1 = {"Info1":{"Module":"Lagorex","Version":"12.5.0(tasmota)","FallbackTopic":"cmnd/DVES_motas/"}}
10:26:32.027 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-7EB07A-4218","IPAddress":"192............
10:26:32.029 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":11}}
10:26:32.031 RSL: RESULT = {"POWER":"ON"}
10:26:32.032 RSL: POWER = ON
10:26:35.415 QPC: Reset
10:26:36.411 RSL: STATE = {"Time":"2023-06-23T10:26:36","Uptime":"0T00:00:09","UptimeSec":9,..................
10:26:36.417 RSL: SENSOR = {"Time":"2023-06-23T10:26:36","LK13BE":{"energy_sum":0.000,"power":0,"meter_number":""}}
10:28:28.381 Script: performs "restart 1"
10:28:28.385 RSL: RESULT = {"Restart":"Restarting"}
10:28:29.390 Script: performs "restart 1"
10:28:29.394 RSL: RESULT = {"Restart":"Restarting"}
10:28:30.390 Script: performs "restart 1"
10:28:30.394 RSL: RESULT = {"Restart":"Restarting"}
10:28:31.394 Script: performs "restart 1"
10:28:31.397 RSL: RESULT = {"Restart":"Restarting"}
10:28:32.411 Script: performs "restart 1"

It seems to have something to do with the scripting.

**Reply by pkkrusty** [reply]
Try `if upsecs==240` instead of >240

**Comment by hw-schrauber** [comment]
Thanks pkkrusty for your brainwork, but why you think that == is more working than >?

I have tested it, and the result is the same. No change about the problem.

I have also restart 99 instead restart 1.

This works as I expected. But I do not understand why. Restart 1 restarts the device with the config as it is stored. Restart 99 restarts the device without saving anything first. But what would be saved with restart 1? And why is it saved in the config?  Do you have an explanation for this?

Maybe I misunderstood or misinterpreted the sentence (1 = restart device with configuration saved to flash) in the command description. I thought the device is started with the data as stored in the config. However, the current status is probably saved in the config before the restart. Thus presumably also upcsec, etc.

So the clean restart is not "restart1". "restart 99" is the command I'm looking for and I have to use!
